haskell-src/lib/ChainwebDb/Types/Common.hs

```diff
@@ -19,18 +19,17 @@ module ChainwebDb.Types.Common where
 import           Data.Text (Text)
 import qualified Data.Text as T
 import           Database.Beam
-import           Database.Beam.AutoMigrate hiding (Table)
 import           Database.Beam.Backend.SQL hiding (tableName)
 import           Database.Beam.Backend.SQL.Row ()
 import           Database.Beam.Backend.SQL.SQL92 ()
 import           Database.Beam.Backend.Types
-import           Database.Beam.Migrate
+import           Database.Beam.Postgres (Postgres)
 ------------------------------------------------------------------------------
 import           ChainwebDb.Types.DbHash
 ------------------------------------------------------------------------------
 
 data ReqKeyOrCoinbase = RKCB_RequestKey (DbHash TxHash) | RKCB_Coinbase
-  deriving (Eq, Ord, Generic)
+  deriving (Eq, Ord, Generic, HasSqlEqualityCheck Postgres)
```

Comment:
This is odd, why do we need this new instance now?

Reply:
It was there before. We need it because of this code
```
exec/Chainweb/Server.hs:419:34: error:
    • No instance for (HasSqlEqualityCheck Postgres ReqKeyOrCoinbase)
        arising from a use of ‘==.’
    • In the first argument of ‘guard_’, namely
        ‘(_ev_requestkey ev ==. val_ (RKCB_RequestKey $ DbHash rk))’
      In a stmt of a 'do' block:
        guard_ (_ev_requestkey ev ==. val_ (RKCB_RequestKey $ DbHash rk))
      In the second argument of ‘($)’, namely
        ‘do ev <- all_ (_cddb_events database)
            guard_ (_ev_requestkey ev ==. val_ (RKCB_RequestKey $ DbHash rk))
            return ev’
    |
419 |        guard_ (_ev_requestkey ev ==. val_ (RKCB_RequestKey $ DbHash rk))
    |                                  ^^^
```

Reply:
What I posted here is the error that results if we get rid of the instance `HasSqlEqualityCheck`.

Reply:
> It was there before.

That's not true though, check out that line on `master`:

https://github.com/kadena-io/chainweb-data/blob/c9b1a026bdec01c739e0763c6a9d97d968a78986/haskell-src/lib/ChainwebDb/Types/Common.hs#L32-L33

And master still has the `_ev_requestkey ev ==. val_ (RKCB_RequestKey $ DbHash rk)` comparison:
https://github.com/kadena-io/chainweb-data/blob/c9b1a026bdec01c739e0763c6a9d97d968a78986/haskell-src/exec/Chainweb/Server.hs#L419

And yet `master` compiles without any issues.

This is no big deal BTW, I'm just trying to understand why removing a bunch of imports could possibly force us to add a new instance

Reply:
What's concerning here is whether this changes the behavior of the equality check. After all, there was no instance before, so the equality comparison was happening one way, and now that GHC is asking for this instance, it will presumably use this derived instance, but are these the same? I imagine that you're working on this branch locally by running `cabal build`, so I wonder whether you're picking up package versions that are different from those that `nix build` uses.

Reply:
Ah OK, I see it now, the instance was declared like this before: https://github.com/kadena-io/chainweb-data/pull/185/files#diff-69d14efae0583571ce896c4c6eba186937d5d7c92c448fa90817689291ddc8d9L48

Mystery solved :)